# Question 2

An experimenter has run a Box-Behnken design and has obtained the results as shown in the table below, where the response variable is the viscosity of a polymer.

In [10]:
# libraries
library(rsm)

x1 <- c(rep(c(-1, 1), 4), rep(0, 7))
x2 <- c(-1, -1, 1, 1, rep(0, 4), rep(c(-1, 1), 2), rep(0, 3))
x3 <- c(rep(0, 4), rep(c(-1, -1, 1, 1), 2), rep(0, 3))
y1 <- c( 535, 580, 596, 563, 645, 458, 350, 600,
         595, 648, 532, 656, 653, 599, 620)
data <- data.frame(x1, x2, x3, y1)
t(data)

[,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10] [,11] [,12] [,13] [,14]
x1  -1    1   -1    1   -1    1   -1    1    0    0     0     0     0     0  
x2  -1   -1    1    1    0    0    0    0   -1    1    -1     1     0     0  
x3   0    0    0    0   -1   -1    1    1   -1   -1     1     1     0     0  
y1 535  580  596  563  645  458  350  600  595  648   532   656   653   599  
   [,15]
x1   0  
x2   0  
x3   0  
y1 620

**(a)** Fit the second-order model.

We fit a second-order model like so:

In [14]:
polymer.quad <- rsm(y1 ~ SO(x1,x2,x3), data = data)
summary(polymer.quad)



Call:
rsm(formula = y1 ~ SO(x1, x2, x3), data = data)

            Estimate Std. Error t value  Pr(>|t|)    
(Intercept)  624.000     18.652 33.4539 4.487e-07 ***
x1             9.375     11.422  0.8208  0.449128    
x2            27.625     11.422  2.4185  0.060228 .  
x3           -26.000     11.422 -2.2763  0.071874 .  
x1:x2        -19.500     16.154 -1.2072  0.281356    
x1:x3        109.250     16.154  6.7632  0.001074 ** 
x2:x3         17.750     16.154  1.0988  0.321915    
x1^2         -75.000     16.813 -4.4608  0.006636 ** 
x2^2          19.500     16.813  1.1598  0.298496    
x3^2         -35.750     16.813 -2.1263  0.086809 .  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Multiple R-squared:  0.945,	Adjusted R-squared:  0.846 
F-statistic: 9.544 on 9 and 5 DF,  p-value: 0.0115

Analysis of Variance Table

Response: y1
                Df Sum Sq Mean Sq F value   Pr(>F)
FO(x1, x2, x3)   3  12216  4072.1  3.9014 0.088473
TWI(x1, x2, x3)  3  50524 16841

**(b)** Perform the canonical analysis. What type of surface has been found?

In [16]:
canonical(polymer.quad)

Near-stationary-ridge situation detected -- stationary point altered
 Change 'threshold' if this is not what you intend



$xs
        x1         x2         x3 
 0.1049878 -0.5526067 -0.1362828 

$eigen
eigen() decomposition
$values
[1]   20.92287    0.00000 -114.69363

$vectors
          [,1]       [,2]       [,3]
x1 -0.02739155  0.5811826  0.8133121
x2  0.99128775 -0.0890686  0.0970329
x3  0.12883440  0.8088842 -0.5736794



Since there is both positive and negative eigenvalues, the surface is a saddlepoint.

**(c)** What operating conditions on $x_1,x_2,$ and $x_3$ maximize viscosity?

In [18]:
ridge <- steepest(polymer.quad, dist = seq(0, 1.412, by = 0.1), descent = FALSE)
ridge

Path of steepest ascent from ridge analysis:


dist x1     x2    x3     | yhat   
1  0.0   0.000 0.000  0.000 | 624.000
2  0.1   0.003 0.083 -0.056 | 627.693
3  0.2  -0.013 0.176 -0.094 | 631.344
4  0.3  -0.032 0.273 -0.119 | 635.216
5  0.4  -0.050 0.372 -0.137 | 639.416
6  0.5  -0.066 0.472 -0.150 | 643.966
7  0.6  -0.081 0.573 -0.158 | 648.892
8  0.7  -0.094 0.674 -0.162 | 654.168
9  0.8  -0.106 0.776 -0.164 | 659.890
10 0.9  -0.117 0.877 -0.164 | 665.948
11 1.0  -0.127 0.979 -0.162 | 672.465
12 1.1  -0.136 1.080 -0.159 | 679.326
13 1.2  -0.144 1.181 -0.155 | 686.594
14 1.3  -0.152 1.282 -0.150 | 694.279
15 1.4  -0.159 1.383 -0.144 | 702.373

The predicted values reach a maximum viscocity of 702.373. The occurs when:  
$$
\begin{align}
x_1 = -0.159 \\
x_2 = 1.383 \\
x_3 = -0.144
\end{align}
$$